src/Compiler/TaskTableView.cpp

```diff
@@ -47,11 +47,27 @@ TaskTableView::TaskTableView(TaskManager *tManager, QWidget *parent)
 
 void TaskTableView::mousePressEvent(QMouseEvent *event) {
   auto idx = indexAt(event->pos());
-  bool expandAreaClicked = expandArea(idx).contains(event->pos());
-  if (expandAreaClicked) {
-    model()->setData(idx, expandAreaClicked, ExpandAreaRole);
+  auto task = m_taskManager->task(idx.row());
+  bool handled = false;
+
+  if (task && idx.data(Qt::DisplayRole).toString() == "Edit settings") {
+    auto parent = task->parentTask();
+    if (parent != nullptr) {
+      // Handle the event first so the selection moves before the modal dialog
+      QTableView::mousePressEvent(event);
+      handled = true;
+      emit TaskDialogRequested(parent->title());
```

Comment:
This was just trying to dynamically work with the existing tableview code. I can look into your suggestions below about a settings type and look into providing a specific category when creating the setting.

Reply:
I'm just curious what happens if we change title? As I understand then we have to change something in the settings and maybe something else. I don't think we have to be depend on the title anywhere. this will cause the problem in the future.

Reply:
If you changed the parent category title and didn't update the settings.json then it would load an empty json object and display an empty dialog. We can either get it all free dynamically and expect the dev to update their meta data or explicitly define lookup values and require the dev to define those at the beginning. Since i'll be creating a TaskType now I can probably associate that Task w/ a fixed string so that it would survive the dev renaming the categories down the road.

Reply:
okay